packages/js/product-editor/src/components/attribute-combobox-field/index.tsx

```diff
@@ -0,0 +1,335 @@
+ */
+interface ComboboxControlProps
+	extends Omit< CoreComboboxControl.Props, 'label' | 'help' > {
+	__experimentalRenderItem?: ( args: {
+		item: ComboboxControlOption;
+	} ) => string | JSX.Element;
+}
+
+/*
+ * Create an alias for the CombobBoxControl core component,
+ * but with the custom ComboboxControlProps interface.
+ */
+const ComboboxControl =
+	CoreComboboxControl as React.ComponentType< ComboboxControlProps >;
+
+type ComboboxControlOptionProps = {
+	item: ComboboxControlOption;
+};
+
+/**
+ * Map the product attribute item to the Combobox core option.
+ *
+ * @param {AttributesComboboxControlItem} attr - Product attribute item.
+ * @return {ComboboxControlOption}               Combobox option.
+ */
+function mapItemToOption(
+	attr: AttributesComboboxControlItem
+): ComboboxControlOption {
+	return {
+		label: attr.name,
+		value: `attr-${ attr.id }`,
+		disabled: !! attr.isDisabled,
+	};
+}
+
+const createNewAttributeOptionDefault: ComboboxControlOption = {
+	label: '',
+	value: '',
+	state: 'draft',
+};
+
+/**
+ * ComboboxControlOption component.
+ *
+ * @param {ComboboxControlOptionProps} props - props.
+ * @return {JSX.Element}                       Component item.
+ */
+function ComboboxControlOption(
+	props: ComboboxControlOptionProps
+): JSX.Element {
+	const { item } = props;
+	if ( item.disabled ) {
+		return <div className="item-wrapper is-disabled">{ item.label }</div>;
+	}
+
+	return <div className="item-wrapper">{ item.label }</div>;
+}
+
+const AttributesComboboxControl: React.FC<
+	AttributesComboboxControlComponent
+> = ( {
+	label,
+	help,
+	current = null,
+	items = [],
+	createNewAttributesAsGlobal = false,
+	instanceNumber = 0,
+	isLoading = false,
+	onChange,
+} ) => {
+	const createErrorNotice = useDispatch( 'core/notices' )?.createErrorNotice;
+	const { createProductAttribute } = useDispatch(
+		EXPERIMENTAL_PRODUCT_ATTRIBUTES_STORE_NAME
+	) as unknown as ProductAttributesActions & WPDataActions;
+
+	const [ createNewAttributeOption, updateCreateNewAttributeOption ] =
+		useState< ComboboxControlOption >( createNewAttributeOptionDefault );
+
+	const clearCreateNewAttributeItem = () =>
+		updateCreateNewAttributeOption( createNewAttributeOptionDefault );
+
+	/**
+	 * Map the items to the Combobox options.
+	 * Each option is an object with a label and value.
+	 * Both are strings.
+	 */
+	const attributeOptions: ComboboxControlOption[] =
+		items?.map( mapItemToOption );
+
+	const options = useMemo( () => {
+		if ( ! createNewAttributeOption.label.length ) {
+			return attributeOptions;
+		}
+
+		return [
+			...attributeOptions,
+			{
+				label:
+					createNewAttributeOption.state === 'draft'
+						? sprintf(
+								/* translators: The name of the new attribute term to be created */
+								__( 'Create "%s"', 'woocommerce' ),
+								createNewAttributeOption.label
+						  )
+						: createNewAttributeOption.label,
+				value: createNewAttributeOption.value,
+			},
```

Comment:
Adding this should depend on if the existing item doesn't exist.
Maybe this should be an option? `allowDuplicates`?

Reply:
Do you mean allowing the creation of a new attribute with the same `name`? 

This is something that the current implementation allows by default. However.

<img width="442" alt="image" src="https://github.com/woocommerce/woocommerce/assets/77539/2ac932a7-56da-4b8d-bb68-7f0b8bf4e6e3">

Above, you can create attributes with the same name by clicking on the `Create "..."` again and again.

 I've just realized that the attributes wp-admin page doesn't when the slug is not explicitly defined:

<img width="658" alt="image" src="https://github.com/woocommerce/woocommerce/assets/77539/1b18f13f-8c91-4559-a123-31abf8956791">

`allowDuplicates` or something more specific to the attribute name. we should create a new issue and add it to the tasks list https://github.com/woocommerce/woocommerce/issues/45294

Reply:
Oooh yeah, I think we do allow this in variation options but not in attributes.
So we did need a way to toggle that.

Reply:
thanks, it makes totally sense

Reply:
https://github.com/woocommerce/woocommerce/issues/46954